test/images/resource-consumer/README.md

```diff
@@ -0,0 +1,66 @@
+# Resource Consumer container for testing
+
+Resource Consumer is a tool for testing kubernetes autoscaling.
+It can help with tests for:
+- cluster size autoscaling
+- horizontal autoscaling of pod - changing the size of replication controller
+- vertical autoscaling of pod - changing its resource limits
+
+image: gcr.io/google_containers/resource_consumer:beta
+
+The container consumes specified amount of resources:
+
+- **CPU** in millicores
+- **Memory** in megabytes
+
+
+Action of consuming resources is send to the container by a POST http request.
+Each http request creates new process.
+It listens on port given as a flag (default 8080).
+Http request handler is in file resource_consumer_handler.go 
+
+**consume CPU http request:**
+
+- suffix "ConsumeCPU"
+- parameters "millicores" and "durationSec"
+
+Consumes specified amount of millicores for durationSec seconds.
+Consume CPU uses "./consume-cpu/consume-cpu" binary (file consume-cpu/consume_cpu.go).
+When CPU consumption is too low this binary uses cpu by calculating math.sqrt(0) 10^7 times
+and if consumption is too high binary sleeps for 10 millisecond.
+
+**consume Memory http request:**
+
+- suffix "ConsumeMem"
+- parameters "megabytes" and "durationSec"
+
+Consumes specified amount of megabytes for durationSec seconds.
+Consume Memory uses stress tool (stress -m 1 --vm-bytes megabytes --vm-hang 0 -t durationSec).
+
+**CURL example**
+$ kubectl run resource-consumer --image=gcr.io/google_containers/resource_consumer:beta
+$ kubectl expose rc resource-consumer --port=8080 --target-port=8080
+$ curl -k -u admin:password --data "millicores=300&durationSec=600" \
```

Comment:
I think it will be more readable to create an external load balancer here (--type=LoadBalancer) and connect directly to the IP of it, w/o using master as proxy. In such case password will not be needed.